### ISSUE TITLE
Namespace is stuck in Activating state on upgraded HA setup 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#20549]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- run Rancher:v2.0.14 in the HA mode 
- add an EKS cluster
- run pre-upgrade automation on the cluster
- upgrade Rancher to v2.0.15-rc5
- run post-upgrade automation on the cluster
- seeing that the automation fails because some namespace is stuck in `Activating` state
- go to the project created by the automation to create a new namespace
- seeing the namespace is stuck in `Activating` state, too

Rancher logs 
```
2019/05/29 17:29:58 [INFO] Creating roleBinding User user-x2ntj Role project-owner
2019/05/29 17:29:59 [INFO] Creating roleBinding User user-x2ntj Role admin
2019/05/29 17:29:59 [INFO] namespaceHandler: addProjectIDLabelToNamespace: adding label field.cattle.io/projectId=p-7ppgg to namespace=fdwf
2019/05/29 17:29:59 [ERROR] namespaceHandler: Sync: error adding project id label to namespace err=Operation cannot be fulfilled on namespaces "fdwf": the object has been modified; please apply your changes to the latest version and try again
2019/05/29 17:29:59 [INFO] namespaceHandler: addProjectIDLabelToNamespace: adding label field.cattle.io/projectId=p-7ppgg to namespace=fdwf
2019/05/29 17:29:59 [ERROR] error updating ns fdwf status: Operation cannot be fulfilled on namespaces "fdwf": the object has been modified; please apply your changes to the latest version and try again
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 
v2.0.14 -> v2.0.15-rc5
- Installation option (single install/HA):
HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Hosted GKE
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.9", GitCommit:"16236ce91790d4c75b79f6ce96841db1c843e7d2", GitTreeState:"clean", BuildDate:"2019-03-25T06:40:24Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.6-eks-d69f1b", GitCommit:"d69f1bf3669bf00b7f4a758e978e0e7a1e3a68f7", GitTreeState:"clean", BuildDate:"2019-02-28T20:26:10Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
(paste the output here)
```

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/17589
https://github.com/rancher/rancher/issues/16674

**Comment by alena1108** [comment]
@cjellick The bug happens because we invoke annotations update in go routine from create lifecycle method that is invoked only once. The create doesn't wait for the return of the go routine results. So we try 10 times, and if all the attempts fail, we never retry:

https://github.com/rancher/rancher/blob/master/pkg/controllers/user/rbac/namespace_handler.go#L66

**Comment by ajfriesen** [comment]
We are creating projects and namespaces via your terraform provider.

When the namespace is staying at activating we get an error from terraform

```
Error: [ERROR] waiting for namespace (cloudwatch-exporter) to be created: timeout while waiting for state to become 'active' (last state: 'activating', timeout: 10m0s)
  on .terraform/modules/cloudwatch-exporter/helm.tf line 1, in resource "rancher2_namespace" "namespace":
   1: resource "rancher2_namespace" "namespace" {
```

This is kind of random as you said above you are just trying 10 times and sometimes it works sometimes it does not.

The real bummer is we need to wait for  10 minutes for the terraform timeout to retry the next random deployment.

**Comment by justcloudnative** [comment]
We, have the same problem

**Comment by gmontoro** [comment]
work around that worked for us in v2.7.6  edit your namespace and remove the following annotation:

    cattle.io/status: '{"conditions": [{"type": "InitialRolesPopulated", "status":
      "Unknown", "message": "Populating initial roles"}]}'

the namespace pops then in Active. 
It would be nice to have this issue fixed instead of working around it :)
